### PR TITLE
Changes to allow data atmosphere with the interspersed sea-ice coupling

### LINKED ISSUE #84: Unable to run with the combined ice-ocean driver
I have been testing the combined ice-ocean driver that was originally implemented in a fully-coupled model with a data atmosphere. A test case of the fully coupled version of this setup can be found here: [MOM6-examples/tree/dev/gfdl/coupled_AM2_LM3_SIS2/Intersperse_ice_1deg](https://github.com/NOAA-GFDL/MOM6-examples/tree/dev/gfdl/coupled_AM2_LM3_SIS2/Intersperse_ice_1deg)

But when trying to run a configuration without an active atmosphere, such as the [Baltic test case](https://github.com/NOAA-GFDL/MOM6-examples/tree/dev/gfdl/ice_ocean_SIS2/Baltic_OM4_025), there is an error in initialization stating `GET_PESET: pelist is not monotonically increasing` and the run fails. 

This is because the sea-ice PEs are separated into slow and fast PEs when the combined-ice ocean driver is used. Fast ice PEs are a subset of the atmosphere PEs and slow ice PEs are the ocean PE list (lines 1364-1373 of coupler_main.F90). However, when using a data atmosphere (`do_atmos=false`), the atmosphere PE list and the ocean PE list are the same (for example [0-15]). This leads to a non-monotonic PE list for the ice PE list which is the union of the fast and slow PE list ( ice PE list would be [0-15 0-15]). To correct this issue, when using the combined ice-ocean driver when the atmosphere and ocean use the same PE list, the ice PE list should be only the fast (or slow) PE list (just [0-15]). 

To reproduce this error,  the combined ice-ocean driver (CIOD) should be turned on in a test case with `do_atmos=false`. The CIOD_input file can be found in the fully coupled interspersed case, and the input.nml file should have the following additions:
```
 &ice_ocean_driver_nml
    output_directory = './',
    parameter_filename = 'CIOD_input',
                         'CIOD_override'
/
```
and in the coupler_nml portion:
```
            do_atmos = .false.,
            do_ice = .true.,
            do_ocean = .true.,
            use_lag_fluxes=.false.
            concurrent = .false.
            concurrent_ice = .true.,
            slow_ice_with_ocean = .true.,
            combined_ice_and_ocean = .true.
```

### PR DESCRIPTION
**Pull request by theresa-morrison**
In the current form, the coupler does not correctly assign PEs when using a data atmosphere (do_atmos=false) and the combined ice-ocean driver (combined_ice_and_ocean = .true.). In that case, all PEs are atmosphere PEs, ocean PEs, slow-ice PEs, and fast-ice PEs. To address this, new options have been added where the ice PE list is assigned.

The logic where the ice PE list is defined has been changed to use the do_atmos, shared_slow_fast_PEs, and slow_ice_with_ocean options for four cases:
 (1) fully coupled configurations that do not use the combined ice-ocean driver,
 (2) fully coupled configurations that do use the combined ice-ocean driver, 
 (3) data atmosphere configurations that do not use the combined ice-ocean driver, and 
 (4) data atmosphere configurations that do use the combined ice-ocean driver. 
Previously, only the shared_slow_fast_PEs option was needed to define the sea ice PE list. 

To confirm this change is successful, the root PE now writes both the slow and fast ice PE list when using the concurrent_ice option. (Added lines 1479-1505).

Finally, changes were made where the logic assumes that a PE is either a slow and fast ice PE but could not be both. 

Fixes #84

### CODE REVIEW

**Comment by thomas-robinson** [comment]
@theresa-morrison can you open an issue related to this PR and describe how to reproduce the current problem and the fix you've proposed?

**Comment by theresa-morrison** [comment]
Yes, I added a new issue explaining the problem.

**Comment by nikizadehgfdl** [comment]
Has any other coupled model tested with this change? E.g., CM4 or OM4?

**Comment by theresa-morrison** [comment]
I have not tested this with CM4 but I have tested it with the versions of OM4 that are in the MOM6-examples. I will try to setup and run CM4 with this version of the coupler.

**Comment by nikizadehgfdl** [comment]
I ran regression tests for ESM4/OM4/AM4 with this PR and they passed. 
I have no idea how to test it for its own purposes. You may want to provide an xml or workdir to be added to the regression suite.

**Comment by thomas-robinson** [comment]
@nikizadehgfdl did you get any timing comparisons?

**Comment by theresa-morrison** [comment]
@nikizadehgfdl  We are working towards adding a case to MOM6-examples that will required this addition to the coupler to work. You can find the draft PR here: [https://github.com/NOAA-GFDL/MOM6-examples/pull/393](https://github.com/NOAA-GFDL/MOM6-examples/pull/393)

**Review by bensonr — APPROVED** [review]

File: full/coupler_main.F90 [hunk 1]
```diff
@@ -1340,40 +1340,71 @@ subroutine coupler_init
...
+        Ice%pelist(:) = Ice%fast_pelist(:)
+        allocate( Ice%slow_pelist(ice_npes) )
+        Ice%slow_pelist(:) = Ice%fast_pelist(:)
+        if(ice_npes .GE. ocean_npes) then
+           allocate(slow_ice_ocean_pelist(ice_npes))
+           slow_ice_ocean_pelist(:) = Ice%slow_pelist(:)
+        else
+           allocate(slow_ice_ocean_pelist(ocean_npes))
+           slow_ice_ocean_pelist(:) = Ocean%pelist(:)
+        endif
```
Comment (bensonr) [inline full/coupler_main.F90:1405]: I'm not going to hold up the approval based on this case, but is this a scenario the ocean group actually runs?

**Pull request merged**